dot/build_spec_test.go

```diff
@@ -2,3 +2,283 @@
 // SPDX-License-Identifier: LGPL-3.0-only
 
 package dot
+
+import (
+	"errors"
+	"github.com/stretchr/testify/require"
+	"os"
+	"testing"
+
+	"github.com/ChainSafe/gossamer/lib/genesis"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestBuildSpec_ToJSON(t *testing.T) {
+	type fields struct {
+		genesis *genesis.Genesis
+	}
+	tests := []struct {
+		name   string
+		fields fields
+		want   []byte
+		err    error
+	}{
+		{
+			name:   "name test",
+			fields: fields{genesis: &genesis.Genesis{Name: "test"}},
+			want: []byte{123, 10, 32, 32, 32, 32, 34, 110, 97, 109, 101, 34, 58, 32, 34, 116, 101, 115, 116, 34,
```

Comment:
maybe put the `want` as type `string` and put the json blob in here? It would make the tests easier to read.

Reply:
Good idea, updated.